tools/Az.Tools.Predictor/Az.Tools.Predictor/AzPredictor.cs

```diff
@@ -194,34 +261,50 @@ public void StartEarlyProcessing(string clientId, IReadOnlyList<string> history)
                                 }
                                 finally
                                 {
-                                    _telemetryClient.OnRequestPrediction(new RequestPredictionTelemetryData(clientId, lastTwoMaskedCommands,
+                                    await localCommandLineExecutedCompletion.Task;
+                                    _telemetryClient.RequestId = requestId;
+                                    _telemetryClient.OnRequestPrediction(new RequestPredictionTelemetryData(client, lastTwoMaskedCommands,
                                                 hasSentHttpRequest,
                                                 (exception is OperationCanceledException ? null : exception)));
                                 }
                             }, _predictionRequestCancellationSource.Token);
                 }
             }
+        }
 
-            (CommandAst Ast, string MaskedValue, bool IsSupported) GetAstAndMaskedCommandLine(string commandLine)
+        public void OnCommandLineExecuted(PredictionClient client, string commandLine, bool success)
+        {
+            if (success && (commandLine.Equals("Connect-AzAccount", StringComparison.InvariantCultureIgnoreCase)
+                        || commandLine.Equals("Clear-AzContext", StringComparison.InvariantCultureIgnoreCase)
+                        || commandLine.Equals("Disconnect-AzAccount", StringComparison.InvariantCultureIgnoreCase)
+                        || commandLine.Equals("Import-AzContext", StringComparison.InvariantCultureIgnoreCase)
+                        || commandLine.Equals("Remove-AzContext", StringComparison.InvariantCultureIgnoreCase)
+                        || commandLine.Equals("Set-AzContext", StringComparison.InvariantCultureIgnoreCase)))
```

Comment:
How about having a hashset of those cmdlet names and use `hashset.Contains(commandline)`? That would be more performant.